packages/flutter_tools/lib/src/flutter_manifest.dart

```diff
@@ -151,13 +152,27 @@ class FontAsset {
   String toString() => '$runtimeType(asset: ${assetUri.path}, weight; $weight, style: $style)';
 }
 
-Future<bool> _validate(Object manifest) async {
-  final String schemaPath = fs.path.join(
+@visibleForTesting
+String buildSchemaDir(FileSystem fs){
+  return fs.path.join(
     fs.path.absolute(Cache.flutterRoot), 'packages', 'flutter_tools', 'schema',
+  );
+}
+
+@visibleForTesting
+String buildSchemaPath(FileSystem fs){
+  return fs.path.join(
+    fs.path.absolute(buildSchemaDir(fs)),
     'pubspec_yaml.json',
   );
-  final Schema schema = await Schema.createSchemaFromUrl(fs.path.toUri(schemaPath).toString());
+}
+
+Future<bool> _validate(Object manifest) async {
+  final String schemaPath = buildSchemaPath(fs);
 
+  final String schemaData = fs.file(schemaPath).readAsStringSync();
```

Comment:
Why was the change in how we load the schema necessary?

Reply:
I couldn't find a way for the previous version to load the path from `MemoryFileSystem` on tests that used it. I believe it should work in production, but I wouldn't have a way to verify the windows behavior. 

If I remember correctly this was the line that caused problems `return new File(uri.scheme == 'file' ? uri.toFilePath() : schemaUrl)`, from `Schema.createSchemaFromUrl`. 

I did a lot of refactors from the 1st working version to the present cleaned one, but couldn't figure out a way to make `Schema.createSchemaFromUrl` work without changing it.

Reply:
BTW the tests I added on `flutter_manifest_test.dart` deal directly with this change.

Reply:
Ah, I see now.  Yeah, it's pretty tricky -- you have to use `IOOverrides` when you want to inject your `MemoryFileSystem` into a package that doesn't use `package:file`.

Check out the following sample for how to do this in a test:

https://github.com/flutter/flutter/blob/f9bb4289e9fd861d70ae78bcc3a042ef1b35cc9d/packages/flutter_test/test/goldens_test.dart#L37-L57

With something like that, you should be able to keep the old call to `Schema.createSchemaFromUrl`, which should yield better asynchrony.

Reply:
There's a modification necessary on Schema (`schema.dart`) to make it work.

Consider the following method `futureSchema`, heavily based on `Schema.createSchemaFromUrl`: 

```
Future<Schema>  futureSchema(String schemaUrl) async {

  if (true || uri.scheme == 'file' || uri.scheme == '') {
    return new io.File(uri.scheme == 'file' ? uri.toFilePath() : schemaUrl)
        .readAsString()
        .then((text) => Schema.createSchema(convert.json.decode(text)));
  } else {
    throw new Exception('Nonsense c: is a valid uri scheme: $schemaUrl');
  }
}
```
Then we can keep the `_validate` method even simpler. 
```
Future<bool> _validate(Object manifest) async {
  final String schemaPath = buildSchemaPath(fs);
  final Schema schema = await futureSchema(schemaPath);
  final Validator validator = new Validator(schema);
  . . .
```

So far I couldn't find any combination of File,`new Uri`, `fs.path`, `fs.file` or `'file:// + ???` that could build a path that can be acceptable to the current version of `createSchemaFromUrl`. `MemoryFileSystem` is ok with `C:\....\....yaml` but the current checks (`if (uri.scheme == 'file' || uri.scheme == '')`) prevent it from working.

Reply:
here's another variation on `createSchemaFromUrl` that apparently would make it work: 
`final Uri uri = Uri.parse(r'$schemaUrl');`.

This whole effort is going beyond my reasonable energy.